### DISCUSSION TITLE
How to setup database through Alembic revision for pytest

### DISCUSSION THREAD

**Discussion by Nannigalaxy** [miguelgrinberg/Flask-Migrate#470] [Q&A]
I need to alembic migration for executing modified SQL queries instead of `db.create_all()`

Here's what I tried
```
@pytest.fixture(scope="session", autouse=True)
def test_database(test_app):
    flask_migrate.upgrade()
    yield


@pytest.fixture(scope="session", autouse=True)
def create_sample_dataset(test_database):
    from tests.utils import PrePopulateDBForTest

    PrePopulateDBForTest().create()
    return
```

But `flask_migrate.upgrade()` gets executed but it does not commit to database.  
Any reason why it is not being committed?

**Comment by miguelgrinberg** [comment]
Do you have a complete example that demonstrates the problem? Flask-Migrate doesn't really do any of this, it just calls Alembic to do the work. Even if this was a bug, it wouldn't be in Flask-Migrate.
